src/runtime/link-accounts-flow.js

```diff
@@ -237,26 +234,41 @@ export class LinkSaveFlow {
   }
 
   /**
-   * @return {?Promise}
-   * @package Visible for testing.
-   */
-  whenLinked() {
-    return this.linkedPromise_;
-  }
-
-  /**
-   * @return {?Promise<boolean>}
-   * @package Visible for testing.
+   * @private
    */
-  whenConfirmed() {
-    return this.confirmPromise_;
+  complete_() {
+    this.dialogManager_.completeView(this.activityIframeView_);
   }
 
   /**
+   * @param {!Object} result
+   * @return {!Promise<boolean>}
    * @private
    */
-  complete_() {
-    this.dialogManager_.completeView(this.activityIframeView_);
+  handleLinkSaveResponse_(result) {
+    // This flow is complete
+    this.complete_();
+    let startPromise;
+    let linkConfirm = null;
+    if (result['linked']) {
+      // When linking succeeds, start link confirmation flow
+      this.dialogManager_.popupClosed();
+      this.deps_.callbacks().triggerFlowStarted(
+          SubscriptionFlows.LINK_ACCOUNT);
+      linkConfirm = new LinkCompleteFlow(this.deps_, result);
+      startPromise = linkConfirm.start();
+    } else {
+      startPromise = Promise.reject(
+          createCancelError(this.win_, 'not linked'));
+    }
+    const completePromise = startPromise.then(() => {
+      this.deps_.callbacks().triggerLinkProgress();
+      return linkConfirm.whenComplete();
+    });
+
+    return completePromise.then(() => {
+      return true;
```

Comment:
why does this need to return true? why not just completePromise?

Reply:
when complete Promise is done, we know that the process succeeded. When something fails, we check if its due to cancellation from the user at any point in the process and we return false. otherwise we let the error be thrown to the caller.

The expected outcome is that upon calling this API, the caller gets a true or false upon completion or error.